src/Backend/SwitcherooControl.vala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2020 Bastien Nocera
+ * Copyright 2020-2021 Justin Haygood
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+[DBus (name = "net.hadess.SwitcherooControl")]
+public interface SwitcherooControlDBus : Object {
+    [DBus (name = "HasDualGpu")]
+    public abstract bool has_dual_gpu { owned get; }
+
+    [DBus (name = "GPUs")]
+    public abstract HashTable<string,Variant>[] gpus { owned get; }
+}
+
+public class Slingshot.Backend.SwitcherooControl : Object {
+
+    private SwitcherooControlDBus dbus { private set; private get; }
+
+    construct {
```

Comment:
```suggestion
    private static SwitcherooControlDBus dbus { private set; private get; }

    static construct {
```

This should probably be a static member of the class, or else we'll end up constructing multiple dbus connections for each instance of the class.